filters/auth/grantconfig.go

```diff
@@ -91,6 +92,11 @@ type OAuthConfig struct {
 	// encrypted access token after a successful token exchange.
 	TokenCookieName string
 
+	// TokenCookieRemoveSubdomains sets the number of subdomains to remove from
+	// the callback request hostname to obtain token cookie domain.
+	// Init converts default nil to 1.
+	TokenCookieRemoveSubdomains *int
```

Comment:
To stay backwards-compatible I had to use pointer here to tell zero and default value (one) apart.
The proper config initialization requires `Init()` call.

If we can sacrifice a bit of compatibility for library users then we could use just `int` and rely on the default skipper flag value of 1.

Reply:
There are not many options here https://stackoverflow.com/questions/68800319/how-to-differentiate-int-null-and-defaulted-to-zero-from-int-actually-equal-to-z

We could also create a wrapper type but that feels too much.

Reply:
If you need this 3 state: 0 value, 0, N, then ptr is the way to go